time.js

```diff
@@ -8,21 +8,321 @@
+                throw Error('"' + str + '" cannot be parsed into a format that can be converted to a ZonedDateTime');
+            }
+        }
+    }
+
+}
+
+/**
+ * Converts the state of the passed in Item to a time.ZonedDateTime
+ * @param {items.Item} item 
+ * @returns {time.ZonedDateTime}
+ * @throws error if the Item's state is not supported or the Item itself is not supported
+ */
+const convertItem = function(item) {
+
+
+    // Uninitialized
+    if(item.isUninitialized) {
+        throw Error('Item ' + item.name + ' is NULL or UNDEF, cannot convert to a time.ZonedDateTime');
+    }
+
+    // Number type Items
+    else if(item.rawState instanceof DecimalType) {
+        return addMillisToNow(item.rawState.floatValue());
+    }
+
+    // String type Items
+    else if(item.rawState instanceof StringType) {
+        return parseString(item.state);
+    }
+
+    // DateTime Items
+    else if(item.rawState instanceof DateTimeType) {
+        return javaZDTtoZDT(item.rawState.getZonedDateTime());
+    }
+
+    // Number:Time type Items
+    else if(item.rawState instanceof QuantityType) {
+        return addQuantityType(item.rawState);
+    }
+
+    else {
+        throw Error(item.type + ' is not supported for conversion to time.ZonedDateTime');
+    }
+}
+
+/**
+ * Converts the passed in when to a time.ZonedDateTime based on the following 
+ * set of rules.
+ * 
+ * - null, undefined: time.ZonedDateTime.now()
+ * - time.ZonedDateTime: unmodified
+ * - Java ZonedDateTime, DateTimeType: converted to time.ZonedDateTime equivalnet
+ * - JavaScript native Date: converted to a time.ZonedDateTime using SYSTEM as the timezone
+ * - number, bigint, Java Number, DecimalType: rounded and added to time.ZonedDateTime.now() as milliseconds
+ * - QuantityType: if the units are Time, added to now
+ * - Item: converts the state of the Item based on the *Type rules described here
+ * - String, Java String, StringType: Parsed based on the following rules
+ *     - ISO 8601: Not yet implemented
+ *     - RFC (output from a Java ZonedDateTime.toString()): parsed to time.ZonedDateTime
+ *     - HH:mm[:ss] (i.e. 24 hour time): that time with today's date (seconds are optional)
+ *     - KK:mm[:ss][ ][aa] (i.e. 12 hour time): that time with today's date (seconds and space between time and am/pm are optional)
+ *     - Duration: any duration string supported by time.Duration (e.g. 'PT5H4M3.210S'), see https://js-joda.github.io/js-joda/class/packages/core/src/Duration.js~Duration.html#static-method-parse
+ * @param {*} when any of the types discussed above
+ * @returns time.ZonedDateTime
+ * @throws error if the type, format, or contents of when are not supported
+ */
+time.toZDT = function(when) {
+
+    // If when is not supplied or null, return now
+    if(when === undefined || when === null) {
+        return time.ZonedDateTime.now();
+    }
+
+    // Pass through if already a time.ZonedDateTime
+    else if(when instanceof time.ZonedDateTime) {
```

Comment:
```suggestion
    if(when instanceof time.ZonedDateTime) {
```
since each of these `if` statements return if matched, there no need for `else` in any of them

Reply:
I wasn't sure. I've used StandardJS as a style guide but it seemed to be silent on this topic. I don't have an opinion either way so have no problem removing the unnecessary else statements.

Reply:
Done